src/Container.ts

```diff
@@ -89,8 +92,16 @@ export class Container {
             params.unshift(null);
         }
 
-        // create a new instance of the requested object
-        const objectInstance = new (type.bind.apply(type, params))();
+        let objectInstance: any;
+
+        if (factory) {
+            // Calling factory function if specified to create an instance.
+            objectInstance = factory.apply(undefined, params);
```

Comment:
I cant answer your question because I don't have examples. These cases come unexpectedly. There are real-life situations when you are using ...params and some day you are realizing you need to pass some option-x and you can't do it at all because of design you choose. Okay, lets go with ...params syntax